components/eamxx/src/python/pyparamlist.hpp

```diff
@@ -7,19 +7,54 @@
 #include <pybind11/numpy.h>
 #include <pybind11/stl.h>
 
+#include <functional>
+
 namespace scream {
 
 struct PyParamList {
   ekat::ParameterList pl;
+  std::reference_wrapper<ekat::ParameterList> pl_ref;
+
+  PyParamList (ekat::ParameterList& src)
+   : pl_ref(src)
+  {}
 
-  PyParamList(const pybind11::dict& d) {
+  PyParamList(const pybind11::dict& d)
+   : PyParamList(d,"")
+  {}
+
+  PyParamList(const pybind11::dict& d, const std::string& name)
+   : pl(name)
+   , pl_ref(pl)
+  {
     parse_dict(d,pl);
   }
 
-  PyParamList(const pybind11::dict& d, const std::string& name) {
-    pl.rename(name);
-    parse_dict(d,pl);
+  PyParamList sublist (const std::string& name) {
+    PyParamList spl(pl.sublist(name));
+    return spl;
+  }
+
+  int get_int (const std::string& name) const {
```

Comment:
Template methods where the template arg cannot be deduced from inputs are not py-interoperable. That is, there is no equivalent of `auto blah = get<int>("my_param");` that you can call from python. Hence, the templated overloads need to have different names.